cmd/podman/common.go

```diff
@@ -67,6 +67,12 @@ func noSubArgs(c *cobra.Command, args []string) error {
 	return nil
 }
 
+func commandRunE() func(*cobra.Command, []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		return errors.Errorf("unrecognized command `%s %s`\nTry '%s --help' for more information.", cmd.CommandPath(), args[0], cmd.CommandPath())
```

Comment:
```
$  ./bin/podman
panic: runtime error: index out of range

goroutine 1 [running]:
main.glob..func57(0x26334a0, 0x27f6770, 0x0, 0x0, 0x0, 0x0)
        /home/esm/go/src/github.com/containers/libpod/cmd/podman/main.go:86 +0xc0
github.com/containers/libpod/vendor/github.com/spf13/cobra.(*Command).execute(0x26334a0, 0xc0000c6170, 0x0, 0x0, 0x26334a0, 0xc0000c6170)
        /home/esm/go/src/github.com/containers/libpod/vendor/github.com/spf13/cobra/command.go:762 +0x473
github.com/containers/libpod/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x26334a0, 0xc, 0x27f8300, 0xc0001bbb00)
        /home/esm/go/src/github.com/containers/libpod/vendor/github.com/spf13/cobra/command.go:852 +0x2fd
github.com/containers/libpod/vendor/github.com/spf13/cobra.(*Command).Execute(0x26334a0, 0xc000058420, 0xc0000bc058)
        /home/esm/go/src/github.com/containers/libpod/vendor/github.com/spf13/cobra/command.go:800 +0x2b
main.main()
        /home/esm/go/src/github.com/containers/libpod/cmd/podman/main.go:226 +0x3c
```

I know my fix was not ideal, but we still need some difference between bad-arg and missing-arg; or at least some way to not dereference a nonexistent index.

Reply:
Also, I know my `badSubCommand` wasn't a great name, but I don't think `commandRunE` really conveys the sense of a function that complains about missing or invalid subarguments; I think a minute or two coming up with a better name would be appreciated by future maintenance programmers.

Reply:
Lets see if this passes tests.